Dockerfile

```diff
@@ -1,38 +1,47 @@
-
-
-FROM ubuntu:21.04 as build
+FROM ubuntu:20.04 as build
 
 ENV LANG C.UTF-8
 ARG DEBIAN_FRONTEND=noninteractive
 
-RUN dpkg --add-architecture i386 && \
-    apt-get update && \
+RUN apt-get update && \
     apt-get upgrade -y && \
     apt-get install -y \
-        binutils:i386 \
-        gcc-10:i386 \
-        g++-10:i386 \
-        python3.10 \
-        python \
+        binutils \
+        gcc-10 \
+        g++-10 \
+	p7zip-full \
+        python3.9 \
         make \
         cmake \
+        curl \
         git \
         lld \
-        libsdl2-dev:i386 \
-        zlib1g-dev:i386 \
-        libbz2-dev:i386 \
-        libpng-dev:i386 \
-        libgles2-mesa-dev && \
-    ln -sf /usr/bin/python3.10 /usr/bin/python3 && \
-    ln -s /usr/bin/gcc-10 /usr/bin/gcc && \
-    ln -s /usr/bin/gcc-10 /usr/bin/cc && \
+        libsdl2-dev \
+        zlib1g-dev \
+        libbz2-dev \
+        libpng-dev \
+        libgles2-mesa-dev && \    
     ln -s /usr/bin/g++-10 /usr/bin/g++ && \
-    ln -s /usr/bin/g++-10 /usr/bin/c++
+    update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-10 10 && \
+    gcc --version && \
+    g++ --version
 
+	
+RUN apt-get clean autoclean && apt-get autoremove --yes && rm -rf /var/lib/apt /var/lib/cache /var/lib/log
+	
 RUN git clone https://github.com/Perlmint/glew-cmake.git && \
     cmake glew-cmake && \
     make -j$(nproc) && \
-    make install ARCH64=false
-
+    make install
+    
+ENV SDL2VER=2.0.22
+RUN curl -sLO https://libsdl.org/release/SDL2-${SDL2VER}.tar.gz && \
+    tar -xzf SDL2-${SDL2VER}.tar.gz && \
+    cd SDL2-${SDL2VER} && \
+    ./configure --prefix=/usr && \
+    make && make install && \
+    rm ../SDL2-${SDL2VER}.tar.gz && \
+    cp -av /lib/libSDL* /lib/x86_64-linux-gnu/
```

Comment:
not opposed to building a newer sdl2 from source, just curious as to what this gets us over 2.0.8 (what's available in 22.04)

Reply:
2.0.14 is needed for the rumble and gyro. If providing SDL, why not the latest. 20.04 comes with 2.0.10, but will cause error building libultraship.

Reply:
sounds good since we're using ubuntu 20.04, my comment only really made sense if we were using a newer distro as the base

i'd still prefer to not need to build it from source but this makes sense for the use case